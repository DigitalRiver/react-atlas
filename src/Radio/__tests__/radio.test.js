import React from "react";
import { mount } from "enzyme";
import { Radio } from "../index";

import renderer from "react-test-renderer";

describe("Test Radio component render", () => {
  it("Render correctly", () => {
    const tree = renderer
      .create(
        <Radio
          label={"Checked Radio"}
          value={"checkedRadio"}
          defaultChecked
          className={"class"}
          title={"TITLE"}
          hidden={false}
          inline
          name={"Mr radio"}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Testing radio component", () => {
  it("Radio - Default checked test", function() {
    const rad = mount(
      <Radio
        checked
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
      />
    );
    expect(rad.props().checked).toEqual(true);
  });

  it("Radio - Simple click test", function() {
    const rad = mount(
      <Radio
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
      />
    );
    rad.simulate("click");
  });

  it("Radio - Simple click test (disabled)", function() {
    const rad = mount(
      <Radio
        disabled
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
      />
    );
    rad.simulate("click");
  });

  it("Radio - Simple click test (with onBeforeChange)", function() {
    const rad = mount(
      <Radio
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
        onClick={function() {
          console.log("onclick");
        }}
        onChange={function() {
          console.log("onchange");
        }}
        onBeforeChange={function() {
          console.log("onbeforechange");
        }}
      />
    );
    rad.simulate("click");
  });

  it("Radio - Simple click test (without onBeforeChange)", function() {
    const rad = mount(
      <Radio
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
        onClick={function() {
          console.log("onclick");
        }}
        onChange={function() {
          console.log("onchange");
        }}
      />
    );
    rad.simulate("click");
  });

  it("Radio - Simple click test (without onBeforeChange) checked", function() {
    const rad = mount(
      <Radio
        checked
        label="Checked Radio"
        value="checkedRadio"
        groupSetChecked={function() {}}
        onClick={function() {
          console.log("onclick");
        }}
        onChange={function() {
          console.log("onchange");
        }}
      />
    );
    rad.simulate("click");
  });
});
