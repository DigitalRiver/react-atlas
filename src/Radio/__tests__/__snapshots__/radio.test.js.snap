// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Radio component render Render correctly 1`] = `
<div
  className="inline_block radio_padding"
  onClick={[Function]}
  style={undefined}
>
  <div
    className="inline_block relative padding"
  >
    <label
      className="class label"
      htmlFor={undefined}
      title="TITLE"
    >
      Checked Radio
    </label>
    <div
      className="float_left"
    >
      <input
        checked={undefined}
        className="input_style"
        defaultChecked={true}
        disabled={false}
        hidden={false}
        id={undefined}
        name="Mr radio"
        onChange={[Function]}
        type="radio"
        value="checkedRadio"
      />
      <div
        className="radio"
      />
    </div>
  </div>
</div>
`;
